depth_segmentation/src/depth_segmentation_node.cpp

```diff
@@ -501,16 +501,26 @@ class DepthSegmentationNode {
       if (!camera_tracker_.getRgbImage().empty() &&
               !camera_tracker_.getDepthImage().empty() ||
           !depth_segmentation::kUseTracker) {
-        computeEdgeMap(depth_msg, rgb_msg, dilated_rescaled_depth, cv_rgb_image,
-                       cv_depth_image, bw_image, mask, &depth_map, &normal_map,
-                       &edge_map);
+        if (params_.dilate_depth_image) {
+          computeEdgeMap(depth_msg, rgb_msg, dilated_rescaled_depth,
+                         cv_rgb_image, cv_depth_image, bw_image, mask,
+                         &depth_map, &normal_map, &edge_map);
+          cv::Mat remove_no_values =
+              cv::Mat::zeros(edge_map.size(), edge_map.type());
+          edge_map.copyTo(remove_no_values,
+                          dilated_rescaled_depth == dilated_rescaled_depth);
+          edge_map = remove_no_values;
+        } else {
```

Comment:
I guess you can, instead of having this if here, add it to the `preprocess` method (https://github.com/ethz-asl/depth_segmentation/blob/master/depth_segmentation/src/depth_segmentation_node.cpp#L385-L391) and just add 
```
else {
    dilated_rescaled_depth = rescaled_depth;
}
```
if the dilation is off. Then you can also remove the lower if you added.